pkg/scheduler/cache/cache.go

```diff
@@ -550,5 +550,11 @@ func (sc *SchedulerCache) Backoff(job *arbapi.JobInfo, event arbcorev1.Event, re
 		return fmt.Errorf("no scheduling specification for job")
 	}
 
+	for _, tasks := range job.TaskStatusIndex {
+		for _, t := range tasks {
+			sc.recorder.Eventf(t.Pod, v1.EventTypeWarning, string(event), reason)
```

Comment:
why do we record event for all task/pods ?

Reply:
This is done in order to match the stock scheduler behaviour which kind of sets the norm in K8s world in terms of users' expectations. No events recorded for Pods which "hang" in `Pending` state for a long time might seem weird ... 
Also this aids in quicker troubleshooting, as one does not need to track down the correct higher level entity to find out why a pod appears to be stuck.